maps/mp/isnipe/extras/topplayer.gsc

```diff
@@ -0,0 +1,147 @@
+#include common_scripts\utility;
+#include maps\mp\_utility;
+#include maps\mp\gametypes\_hud_util;
+
+/**
+ * Initialize the topPlayer array on level.
+ * This is used to store data of the current top player.
+ *
+ * The following variables are stored:
+ * - level.topPlayer["playerName"]: this contains the name
+ *                                  of the current top player
+ * - level.topPlayer["playerGuid"]: this contains the guid
+ *                                  of the current top player
+ * - level.topPlayer["killDeathRatio"]: this contain the k/d ratio
+ *                                      of current top player
+ */
+init() {
+    if (!isDefined(level.topPlayer)) {
+        level.topPlayer = [];
+    }
+
+    _setSettingIfNotExists("playerName", "");
+    _setSettingIfNotExists("playerGuid", "");
+    _setSettingIfNotExists("killDeathRatio", 0);
+
+    self thread _updateRatios();
+}
```

Comment:
While your argument makes sense, this would mean that each player would have this running for them in a seperate thread instead of one global method running for everyone in one thread.